com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -598,7 +600,10 @@ internal void SetNetworkParenting(bool isReparented, ulong? latestParent)
         /// <returns>Whether or not reparenting was successful.</returns>
         public bool TrySetParent(Transform parent, bool worldPositionStays = true)
         {
-            return TrySetParent(parent.GetComponent<NetworkObject>(), worldPositionStays);
+            var networkObject = parent.GetComponent<NetworkObject>();
+
+            // If the parent doesn't have a NetworkObjet then return false, otherwise continue trying to parent
+            return networkObject == null ? false : TrySetParent(networkObject, worldPositionStays);
```

Comment:
You cannot pass a null value in since this overloaded version of TrySetParent is a Transform (struct).
So, the appropriate response here if the transform does not have a NetworkObject is to return false.